engine/core/source/org/pentaho/reporting/engine/classic/core/style/TextRotation.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ *  terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ *  Foundation.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License along with this
+ *  program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ *  or from the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Lesser General Public License for more details.
+ *
+ *  Copyright (c) 2006 - 2016 Pentaho Corporation..  All rights reserved.
+ */
+
+package org.pentaho.reporting.engine.classic.core.style;
+
+import org.pentaho.reporting.engine.classic.core.util.ObjectStreamResolveException;
+
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+
+public class TextRotation implements Serializable {
+  public static final TextRotation D_90 = new TextRotation( "90" );
+  public static final TextRotation D_270 = new TextRotation( "-90" );
+  private static final String CSS =
+    "transform: rotate(#deg); -ms-transform: rotate(#deg); -webkit-transform: rotate(#deg);";
```

Comment:
The use of # as magic constant might store trouble for future maintainer. Replace # with {0} and use "MessageFormat.format(CSS, String.valueOf(getNumericValue()));" and you never have to explain the meaning of # in this context to any other developer.